ostd/src/arch/x86/cpu/mod.rs

```diff
@@ -489,39 +483,27 @@ impl FpuState {
         }
 
         Self {
-            state_area: XSaveArea::init(),
+            state_area: Box::new(XSaveArea::default()),
             area_size,
-            is_valid: AtomicBool::new(true),
         }
     }
 
-    /// Returns whether the instance can contains valid state.
-    pub fn is_valid(&self) -> bool {
-        self.is_valid.load(Relaxed)
-    }
-
-    /// Save CPU's current FPU state into this instance.
-    pub fn save(&self) {
-        let mem_addr = &*self.state_area as *const _ as *mut u8;
+    /// Saves CPU's current FPU state into the given XSaveArea.
+    pub fn save_to(xsave_area: &mut XSaveArea) {
+        let mem_addr = xsave_area as *const _ as *mut u8;
```

Comment:
This pattern is unsound. When converting a reference to a pointer, it's crucial whether the initial pointer is `const` or `mut`. If you intend to modify the underlying memory of the pointer, you should always convert the reference to a mut pointer first. You can view the detailed description
at https://github.com/asterinas/asterinas/pull/1778#issue-2765390329.

So the code should be
```rust
let mem_addr = xsave_area as *mut _ as *mut u8;
```